### PR TITLE
Feature/level selection

### PR DESCRIPTION
**Pull request by UlricVilleneuve**

### CODE REVIEW

File: Eskillate/Assets/Scripts/Core/LoadAdditiveScene.cs [hunk 1]
```diff
@@ -17,6 +17,20 @@ public static void LoadGenericMenus(MonoBehaviour mono)
...
         }
 
+        public static void LoadSceneAdditively(MonoBehaviour mono, string sceneName, Action callback)
+        {
+            mono.StartCoroutine(LoadAdditiveScene.LoadAsync(sceneName, callback));
+        }
+
+        public static void LoadScene(MonoBehaviour mono, string sceneName, Action callback)
+        {
+            var sceneLoadedCallback = new SceneLoadedCallback()
```
Comment (princess-chichie) [inline Eskillate/Assets/Scripts/Core/LoadAdditiveScene.cs]: do you really need the class scene loaded call back?

**Review by princess-chichie — CHANGES_REQUESTED** [review]

File: Eskillate/Assets/Scripts/DragAndDrop/GameController.cs [hunk 2]
```diff
@@ -18,22 +19,74 @@ void Start()
...
+                levelCountForThisLevelTrio++;
+            }
+
+            while(levelCountForThisLevelTrio != 3)
+            {
+                GameObject.Destroy(currentLevelTrio.transform.GetChild(levelCountForThisLevelTrio).gameObject);
+                levelCountForThisLevelTrio++;
+            }
+
+            // Set Y pos to 1 to be at scrolled to the top
```
Comment (UlricVilleneuve) [inline Eskillate/Assets/Scripts/DragAndDrop/GameController.cs]:
```suggestion
            // Set Y pos to 1 to be scrolled to the top
```

File: Eskillate/Assets/Scripts/LowPop/GameController.cs [hunk 3]
```diff
@@ -31,19 +33,72 @@ void Start()
...
+            }
+
+            // Set Y pos to 1 to be at scrolled to the top
+            scrollListContent.transform.parent.transform.parent.GetComponent<ScrollRect>().normalizedPosition = new Vector2(0, 1);
+        }
+
+        private void OnLevelSelected(int levelId)
+        {
+            UnloadLevel();
+            LoadLevel(levelId - 1);
```
Comment (UlricVilleneuve) [inline Eskillate/Assets/Scripts/LowPop/GameController.cs:93]: I dont like this. I think the levels ID should reflect their position in the list.

File: Eskillate/Assets/Scripts/DragAndDrop/ShapeCollision.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
-﻿using UnityEngine;
+﻿using Core;
```
Comment (UlricVilleneuve) [inline Eskillate/Assets/Scripts/DragAndDrop/ShapeCollision.cs:1]: Necessary?

File: Eskillate/Assets/Scripts/DragAndDrop/GameController.cs [hunk 5]
```diff
@@ -18,22 +19,74 @@ void Start()
...
+            scrollListContent.transform.parent.transform.parent.GetComponent<ScrollRect>().normalizedPosition = new Vector2(0, 1);
+        }
+
+        private void OnLevelSelected(int levelId)
+        {
+            UnloadLevel();
+            LoadLevel(levelId-1);
```
Comment (UlricVilleneuve) [inline Eskillate/Assets/Scripts/DragAndDrop/GameController.cs:86]: Same comment as the other file. 
Btw, lots of code repetition, can you find a way to use common code. Maybe that should be done in another PR though.

File: Eskillate/Assets/Scripts/DragAndDrop/ShapeCollision.cs (hunk 4 above)
Comment (UlricVilleneuve) [reply Eskillate/Assets/Scripts/DragAndDrop/ShapeCollision.cs:1]: Yes, because AusioScript is now in Core

**Pull request merged**